packages/resolver/lib/resolver.ts

```diff
@@ -1,28 +1,64 @@
+import debugModule from "debug";
+const debug = debugModule("resolver");
+
 const contract = require("@truffle/contract");
 const expect = require("@truffle/expect");
 const provision = require("@truffle/provisioner");
 
-import { ResolverSource } from "./source";
-import { EthPMv1, NPM, GlobalNPM, FS, Truffle, ABI } from "./sources";
+import { ResolverSource, ResolvedSource } from "./source";
+import { EthPMv1, NPM, GlobalNPM, FS, Truffle, ABI, Vyper } from "./sources";
+
+export interface ResolverOptions {
+  includeTruffleSources?: boolean;
+  translateJsonToSolidity?: boolean;
+  resolveVyperModules?: boolean;
+}
+
+const defaultResolverOptions = {
+  includeTruffleSources: false,
+  translateJsonToSolidity: true,
+  resolveVyperMoudles: false,
+};
 
 export class Resolver {
   options: any;
   sources: ResolverSource[];
 
-  constructor(options: any, includeTruffleSource: boolean = false) {
-    expect.options(options, ["working_directory", "contracts_build_directory"]);
+  constructor(options: any, resolverOptions: ResolverOptions = {}) {
+    expect.options(
+      options,
+      ["working_directory", "contracts_build_directory", "contracts_directory"]
+    );
+
+    resolverOptions = {
+      ...defaultResolverOptions,
+      ...resolverOptions
+    };
+    const {
+      includeTruffleSources,
+      translateJsonToSolidity,
+      resolveVyperModules
+    } = resolverOptions;
 
     this.options = options;
     this.sources = [
+      ...(includeTruffleSources ? [new Truffle(options)] : []),
       new EthPMv1(options.working_directory),
       new NPM(options.working_directory),
       new GlobalNPM(),
-      new ABI(options.working_directory, options.contracts_build_directory),
+      ...(translateJsonToSolidity
+        ? [
+            new ABI(
+              options.working_directory,
+              options.contracts_build_directory
+            )
+          ]
+        : []),
       new FS(options.working_directory, options.contracts_build_directory)
     ];
 
-    if (includeTruffleSource) {
-      this.sources.unshift(new Truffle(options));
+    if (resolveVyperModules) {
```

Comment:
Have you gotten any ideas on how we could address this weird one-off behavior? I worry that code smells are starting to compound with https://github.com/trufflesuite/truffle/issues/2599.

Reply:
I mean the real answer is some sort of mechanism for making resolvers from existing ones, but I haven't seriously thought about how that would work.  The simplest way, I suppose, would be for resolvers to just expose the options they were created with... I mean, should I add that here?  If I do I'll want to go changing other parts of the code to actually make use of this new approach.

Reply:
No definitely any further change is out of scope.